### DISCUSSION TITLE
Idea: Docs for implementing NixOS-WSL in a Flake

### DISCUSSION THREAD

**Discussion by gpfaucher** [nix-community/NixOS-WSL#315] [Ideas]
Hi there! I'm still quite new to Nix and NixOS, but I'm getting started. I've had to start learning about NixOS-WSL due to a new job that will require me to switch from Linux to Windows. I've got everything up and running, but I had one major gripe: the documentation suggests creating a WSL (instance) using a tarball, an for the rest of the documentation imports it as a module. This requires configurations to be build with the `--impure flag` and is a lot less neat that implementing it as a flake IMO.

I'd be more than willing to make a PR for this, as it's quite straightforward: I think adding an example `flake.nix` with ` nixos-wsl.url = "github:nix-community/NixOS-WSL";` as input, and an example `nixosSystem` importing `nixos-wsl.nixosModules.wsl` would solve this.

If I missed something and / or made an error please let me know! I'm still very inexperienced with open source and I'm still a junior / medior developer as well. Cheers!

**Comment by SuperSandro2000** [comment]
sure, why not, we could add an example flake
